consensus/src/lib.rs

```diff
@@ -208,70 +245,39 @@ impl<N: GossipNetwork, S: Storage> Consensus<N, S> {
-                    ConsensusEvent::NonNilPrevote {
-                        proposal: index,
-                        signer,
-                        round: round as usize,
-                        time: timestamp,
-                    }
-                }
-                _ => unimplemented!(),
-            };
-            // Not directly commit the change in case of possible dms errors.
-            let mut consensus_state_copy = self.state.consensus_state.clone();
-            let result = consensus_state_copy.progress(event);
-            if let Some(result) = result {
-                for response in result {
-                    match response {
-                        ConsensusResponse::BroadcastNilPrevote { round } => {
-                            if let Some(private_key) = self.this_node_key.as_ref() {
-                                let message = ConsensusMessage::NilPreVoted(round as u64);
-                                let message = serde_json::to_string(&message).unwrap();
-                                let signature = TypedSignature::sign(&message, private_key)
-                                    .expect("private key already verified");
-                                self.dms
-                                    .add_message(
-                                        network_config,
-                                        known_peers,
-                                        Message::new(message, signature)
-                                            .expect("signature just created"),
-                                    )
-                                    .await?
-                            }
-                            final_result.push(ProgressResult::NilPreVoted(round as u64, timestamp));
-                        }
-                        _ => unimplemented!(),
-                    }
-                }
-            }
-            self.state.consensus_state = consensus_state_copy;
+            let consensus_message = serde_json::from_str::<ConsensusMessage>(message.data())
+                .expect("this must be already verified by the message filter");
+            let consensus_event = self.consensus_message_to_event(&consensus_message, signer);
+            progress_responses.extend(vetomint_copy.progress(consensus_event, timestamp));
         }
-        self.storage
-            .add_or_overwrite_file(STATE_FILE_NAME, serde_json::to_string(&self.state).unwrap())
+        let final_result = self
+            .process_multiple_responses(progress_responses, network_config, known_peers, timestamp)
             .await?;
+        // The change is applied here as we reached here without facing an error.
+        self.state
+            .updated_messages
+            .extend(messages.into_iter().map(|m| m.to_hash256()));
+        self.state.vetomint = vetomint_copy;
+        // Note, Todo: For now, storage errors are not handled.
+        self.commit_state_to_storage().await?;
         Ok(final_result)
```

Comment:
Where is the storage clearing process in this method? Is it in the other related method?
(You said `For the case 4, it will clear the storage` at the documentation.)

Reply:
I think the comment should be correctd. The storage is cleaned when the node creates a new instance of consensus module.

Reply:
corrected.